PWGHF/HFC/Tasks/femtoDreamPairTaskCharmHadrons.cxx

```diff
@@ -357,6 +405,47 @@ struct femtoDreamPairTaskCharmHadrons {
     }
   }
   PROCESS_SWITCH(femtoDreamPairTaskCharmHadrons, processMixedEvent, "Enable processing mixed events", true);
+
+  /// process function for to call doSameEvent with Monte Carlo
+  /// \param col subscribe to the collision table (Monte Carlo Reconstructed reconstructed)
+  /// \param parts subscribe to joined table FemtoDreamParticles and FemtoDreamMCLables to access Monte Carlo truth
+  /// \param FemtoDreamMCParticles subscribe to the Monte Carlo truth table
+  void processSameEventMc(FilteredMcColision& col,
+                          o2::aod::FDMCCollisions&,
+                          FilteredFDMcParts const& parts,
+                          o2::aod::FDMCParticles const&, FilteredCharmMcCands const& candidates)
+  {
+    auto SliceMcTrk1 = PartitionMcTrk1->sliceByCached(aod::femtodreamparticle::fdCollisionId, col.globalIndex(), cache);
+    auto SliceMcCharmHad = PartitionMcHF->sliceByCached(aod::femtodreamparticle::fdCollisionId, col.globalIndex(), cache);
+
+    if (SliceMcTrk1.size() == 0 && SliceMcCharmHad.size() == 0) {
+      return;
+    }
+    doSameEvent<true>(SliceMcTrk1, SliceMcCharmHad, parts, candidates, col);
+  }
+  PROCESS_SWITCH(femtoDreamPairTaskCharmHadrons, processSameEventMc, "Enable processing same event for Monte Carlo", false);
+
+  /// brief process function for to call doMixedEvent with Monte Carlo
+  /// @param cols subscribe to the collisions table (Monte Carlo Reconstructed reconstructed)
+  /// @param parts subscribe to joined table FemtoDreamParticles and FemtoDreamMCLables to access Monte Carlo truth
+  /// @param FemtoDreamMCParticles subscribe to the Monte Carlo truth table
+  void processMixedEventMc(FilteredMcColisions const& cols, o2::aod::FDMCCollisions const&, FilteredFDMcParts const& parts, o2::aod::FDMCParticles const&, FilteredCharmMcCands const& candidates)
```

Comment:
same here about the collision table

Reply:
This has no impact so I removed it for now. Since I was using a single file it would be nice to see the output on full stats